internal/httputil/certpoolwatcher.go

```diff
@@ -126,3 +129,77 @@ func (cpw *CertPoolWatcher) drainEvents() {
 		}
 	}
 }
+
+func logPath(p string, log logr.Logger) {
+	fi, err := os.Stat(p)
+	if err != nil {
+		log.Error(err, "error in os.Stat()", "path", p)
+		return
+	}
+	if !fi.IsDir() {
+		logFile(p, log)
+		return
+	}
+	dirEntries, err := os.ReadDir(p)
+	if err != nil {
+		log.Error(err, "error in os.ReadDir()", "path", p)
+		return
+	}
+	for _, e := range dirEntries {
+		file := filepath.Join(p, e.Name())
+		fi, err := os.Stat(file)
+		if err != nil {
+			log.Error(err, "error in os.Stat()", "file", file)
+			continue
+		}
+		if fi.IsDir() {
+			log.Info("ignoring subdirectory", "directory", file)
+			continue
+		}
+		logFile(file, log)
+	}
+}
+
+func logFile(f string, log logr.Logger) {
+	data, err := os.ReadFile(f)
+	if err != nil {
+		log.Error(err, "error in os.ReadFile()", "file", f)
+		return
+	}
+	var block *pem.Block
+	block, data = pem.Decode(data)
+	if block == nil {
+		log.Error(nil, "no block returned from pem.Decode()", "file", f)
+		return
+	}
+	crt, err := x509.ParseCertificate(block.Bytes)
+	if err != nil {
+		log.Error(err, "error in x509.ParseCertificate()", "file", f)
+		return
+	}
+
+	count := 0
+	for len(data) > 0 {
+		block, data = pem.Decode(data)
+		if block != nil {
+			_, err := x509.ParseCertificate(block.Bytes)
+			if err == nil {
+				count = count + 1
+			}
+		}
+	}
+
+	// Find an appopriate certificate identifier
+	args := []any{"file", f}
+	if s := crt.Subject.String(); s != "" {
+		args = append(args, "subject", s)
+	} else if crt.DNSNames != nil {
+		args = append(args, "DNSNames", crt.DNSNames)
+	} else if s := crt.SerialNumber.String(); s != "" {
+		args = append(args, "serial", s)
+	}
+	if count > 1 {
+		args = append(args, "additional-certs", count)
+	}
+	log.Info("watching certificate", args...)
```

Comment:
Could we log all of the x509 certs in a file instead of just the first?

`pem.Decode`, `x509.ParseCertificate`, and `log.Info("loaded certificate", args)` in a loop?

That might be large, so maybe a "small" log for level 0 and a more thorough log for level 2?

Reply:
👍 for log-level differentiation.

I've also been wondering about how much information should we be logging here at all from security standpoint about the details of the certificates that are being used?

Reply:
I originally had a listing of all certs, but it was quite long... so it's doable. 
Since these are all public certs, what we publish about them doesn't really matter, but I was choosing a minimum amount of information (i.e. subject or AltDNS or serial)